src/models/transfers.js

```diff
@@ -318,6 +318,11 @@ function * fulfillTransfer (transferId, fulfillmentUri) {
   const fulfillment = ConditionFulfillment.fromDataExternal(fulfillmentUri)
   fulfillment.setTransferId(transferId)
   const existingFulfillment = yield db.transaction(function * (transaction) {
+    // Set isolation level to avoid reading "prepared" transaction that is currently being
+    // executed by another request. This ensures the transfer can be fulfilled only once.
+    if (db.client === 'pg' || db.client === 'strong-oracle') {
+      yield transaction.raw('SET TRANSACTION ISOLATION LEVEL SERIALIZABLE')
+    }
```

Comment:
@clark800 I updated this to include oracle. The issue was due to a semi-colon on the end of the SQL string. I removed mysql, since you said your script doesn't support it. I changed the isolation level to `serializable` for postgres because oracle doesn't support `repeatable read`. It only support `serializable` and `read committed`. See the [docs](https://docs.oracle.com/cd/B28359_01/server.111/b28286/statements_10005.htm). I also changed postgres to `serializable` because I think it's confusing to have different isolation levels depending on the database.